test/testProcessEvents.js

```diff
@@ -0,0 +1,132 @@
+    },
+    {
+        fromChainId: ChainId.ARBITRUM,
+        fromChainBlock: 17229003 ,
+        toChainId: ChainId.OPTIMISM,
+        toChainBlock: 14062015,
+        expectedRes: {"fromTxnHash":"0x87ba82a105088172b03a4c51dfe7fc77c3a0d028ac46361f62545892ade6b7b9","toAddress":"0xf51e5a0A85d29AA0508894405D734BcF044dcb5b","fromAddress":"0xf51e5a0A85d29AA0508894405D734BcF044dcb5b","sentValue":"100000000000000000","sentValueFormatted":0.1,"fromChainId":42161,"toChainId":10,"fromChainBlock":17229003,"sentTime":1657253121,"sentTokenSymbol":"WETH","sentTokenAddress":"0x82aF49447D8a07e3bd95BD0d56f35241523fBab1","kappa":"0xbad4436b75bd5e4bc801e5eadaefab23d251e469ffc8b6f0eee5bf264697a846","pending":false,"__v":0,"receivedTime":1657253251,"receivedTokenAddress":"0x121ab82b49B2BC4c7901CA46B8277962b4350204","receivedTokenSymbol":"WETH","receivedValue":"96973053038470143","receivedValueFormatted":0.0969730530384701,"swapSuccess":true,"toChainBlock":14062015,"toTxnHash":"0x8134a3b9a55d5614b062030ea4901e39df11ca1640abe3bf8041d8256e8b1e73"}
+    },
+    {
+        fromChainId: ChainId.FANTOM,
+        fromChainBlock: 41149116 ,
+        toChainId: ChainId.POLYGON,
+        toChainBlock: 29904168,
+        expectedRes: {"fromTxnHash":"0x282d41ac201664c7419eedf58a36a990adbc81242eb0c12f8af473aee6e8f1fd","toAddress":"0xC6D922d34b354F949EF992eeFacE2F6F7a06fB29","fromAddress":"0x935b702Da05d8701AF3a6101166c7895ff5b3741","sentValue":"998882534","sentValueFormatted":998.882534,"sentValueUSD":998.552184986889,"fromChainId":250,"toChainId":137,"sentTime":1655982521,"sentTokenSymbol":"USDC","sentTokenAddress":"0x04068DA6C83AFCFA0e13ba15A6696662335D5B75","kappa":"0xfc2c4d7d2d8b7674daede0446f479a33c1ff07aad911ac36cd51c863dbb41219","pending":false,"__v":0,"receivedTime":1655982563,"receivedTokenAddress":"0x2791Bca1f2de4661ED88A30C99A7a9449Aa84174","receivedTokenSymbol":"USDC","receivedValue":"997959253","receivedValueFormatted":997.959253,"receivedValueUSD":997.62920933307,"swapSuccess":true,"toTxnHash":"0x0c21f80c373139ea1157baf3fce32b33f8c764814b8915a9931a7eb872a85527","fromChainBlock":41149116,"toChainBlock":29904168}
+    }
+]
+
+async function processEventForChainBlock(chainId, chainBlock) {
+    let chainConfig = ChainConfig[chainId]
+    let bridgeContractAddress = ethers.utils.getAddress(chainConfig.bridge);
+    let w3Provider = getW3Provider(chainConfig.id);
+
+    let bridgeContract = buildBridgeContract(
+        bridgeContractAddress,
+        getBridgeContractAbi(),
+        w3Provider
+    )
+    let events = await bridgeContract.queryFilter(
```

Comment:
we need to test rpc failures here to make sure stuff gets retried

Reply:
Makes sense. I'll actually have to rewrite some parts of the app that address RPC failures, not sure how easy it would be to test them, though I can do some mocking